ipfs-api-prelude/src/request/pin.rs

```diff
@@ -46,3 +46,132 @@ pub struct PinRm<'a> {
 impl<'a> ApiRequest for PinRm<'a> {
     const PATH: &'static str = "/pin/rm";
 }
+
+#[derive(Serialize)]
+pub struct PinRemoteAdd<'a> {
+    #[serde(rename = "arg")]
+    pub key: &'a str,
+
+    pub service: Option<&'a str>,
+    pub name: Option<&'a str>,
+    pub background: Option<bool>,
+}
+
+impl<'a> ApiRequest for PinRemoteAdd<'a> {
+    const PATH: &'static str = "/pin/remote/add";
+}
+
+#[derive(Serialize)]
+pub struct PinRemoteLs<'a> {
+    pub service: Option<&'a str>,
+    pub name: Option<&'a str>,
+    pub cid: Option<Cids<'a>>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialize_pin_status"
+    )]
+    pub status: Option<&'a [PinStatus]>,
+}
+
+impl<'a> ApiRequest for PinRemoteLs<'a> {
+    const PATH: &'static str = "/pin/remote/ls";
+}
+
+#[derive(Serialize)]
+pub struct PinRemoteRm<'a> {
+    pub service: Option<&'a str>,
+    pub name: Option<&'a str>,
+    pub cid: Option<Cids<'a>>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "serialize_pin_status"
+    )]
+    pub status: Option<&'a [PinStatus]>,
+    pub force: Option<bool>,
+}
+
+impl<'a> ApiRequest for PinRemoteRm<'a> {
+    const PATH: &'static str = "/pin/remote/rm";
+}
+
+pub struct Cids<'a>(&'a [&'a str]);
+
+impl<'a> From<&'a [&'a str]> for Cids<'a> {
+    fn from(data: &'a [&'a str]) -> Self {
+        Self(data)
+    }
+}
+
+impl<'a> Serialize for Cids<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let cids = self.0.join(",");
+        serializer.serialize_str(&cids)
+    }
+}
+
+#[derive(Serialize)]
+pub enum PinStatus {
+    Queued,
+    Pinning,
+    Pinned,
+    Failed,
+}
+
+impl Default for PinStatus {
+    fn default() -> Self {
+        Self::Pinned
+    }
+}
+
+fn serialize_pin_status<S>(
+    pin_status: &Option<&[PinStatus]>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let pin_status = pin_status.unwrap();
+    let pin_status = pin_status
+        .iter()
+        .map(|item| match *item {
```

Comment:
Implement `Display` for `PinStatus` maybe.

Reply:
Just implemented Display, it looks like we need to use `collect::<Vec<String>>()` instead of `collect::<Vec<&str>>()`, I wonder if this will cause an extra memory allocation operation